iOSDFULibrary/Classes/Utilities/DFUPackage/Manifest/SoftdeviceBootloaderInfo.swift

```diff
@@ -30,17 +30,47 @@
 
 import Foundation
 
-class SoftdeviceBootloaderInfo: ManifestFirmwareInfo {
-    var blSize: UInt32 = 0
```

Comment:
Those fields (`blSize` and `sdSize`) are required for Legacy DFU and cannot be removed. They can changed to optional instead of default to 0, but Legacy DFU needs to send 0 when needed.

Reply:
Here's a valid JSON file for Legacy DFU:
```json
{
    "manifest": {
        "softdevice_bootloader": {
            "bin_file": "sd_bl.bin",
            "bl_size": 14284,
            "dat_file": "sd_bl.dat",
            "init_packet_data": {
                "application_version": 4294967295,
                "device_revision": 65535,
                "device_type": 65535,
                "firmware_crc16": 25444,
                "softdevice_req": [
                    65534
                ]
            },
            "sd_size": 108588
        },
        "application": {
            "bin_file": "dfu_test_app_hrm_s132.bin",
            "dat_file": "dfu_test_app_hrm_s132.dat",
            "init_packet_data": {
                "application_version": 65535,
                "device_revision": 65535,
                "device_type": 65535,
                "firmware_crc16": 35139,
                "softdevice_req": [
                    109
                ]
            }
        }
    }
}
```

And here for Secure DFU:
```json
{
    "manifest": {
        "application": {
            "bin_file": "app.bin",
            "dat_file": "app.dat"
        },
        "softdevice_bootloader": {
            "bin_file": "sd_bl.bin",
            "dat_file": "sd_bl.dat",
            "info_read_only_metadata": {
                "bl_size": 24308,
                "sd_size": 152984
            }
        }
    }
}
```